Add Documentation for IML APIs. (#471)
heremaps/here-data-sdk-typescript: docs/work-with-interactive-map-layer.md

      For more information, see [Searching for features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-search.html) in the Data API Guide.

    - To get features using iteration, call the `iterateFeatures` function.

      You get an ordered set of features, and none of them is returned twice. For more information, see [Iterating features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-iterate.html) in the Data API Guide.

    - To get features using statistics, call the `getStatistics` function with the ID of the layer.

      ```typescript
      const result = await InteractiveApi.getStatistics(requestBuilder, {
        layerId: "your-layer-id",
      });
      ```

## Update data in and publish it to an interactive map layer

You can update existing features in an interactive map layer or publish new ones.

**To update and publish features:**

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

2. Call the `patchFeature`, `postFeatures`, `putFeature`, or `putFeatures`  functions.

>     For more information, see [Update data in an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/updating-data-interactive.html) and [Publish to an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/publishing-data-interactive.html) in the Data API Guide.

Reviewer: Please add a blank line before 184.

## Delete data from an interactive map layer

You can delete features from an interactive map layer when you no longer need them.

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

2. Call the `deleteFeature` or `deleteFeatures` functions.

>     For more information, see [Delete data from an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/deleting-data-interactive.html) in the Data API Guide.

Reviewer: Please add a blank line before 195.

     ```typescript
     const result = await InteractiveApi.getStatistics(requestBuilder, {
       layerId: "your-layer-id",
     });
     ```

## Update data in and publish it to an interactive map layer

You can update existing features in an interactive map layer or publish new ones.

**To update and publish features:**

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

2. Call the `patchFeature`, `postFeatures`, `putFeature`, or `putFeatures` functions.

   For more information, see [Update data in an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/updating-data-interactive.html) and [Publish to an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/publishing-data-interactive.html) in the Data API Guide.

## Delete data from an interactive map layer

You can delete features from an interactive map layer when you no longer need them.

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

> 2) Call the `deleteFeature` or `deleteFeatures` functions.

Reviewer: ```suggestion
2. Call the `deleteFeature` or `deleteFeatures` functions.
```

   ```typescript
   const result = await InteractiveApi.getFeatures(requestBuilder, {
     id: "your-feature1-id,your-feature2-id",
     layerId: "your-layer-id",
   });
   ```

   Note: Requesting multiple features will always return a `FeatureCollection`, even if the request returns only one or no feature.

**To get features by bounding box from the interactive map layer:**

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

2. Call the `getFeaturesByBBox` function with the IDs of the features that you request and layer ID.

   ```typescript
   const result = await InteractiveApi.getFeaturesByBBox(requestBuilder, {
     bbox: "13.082,52.416,13.628,52.626", // Bounding box of Berlin
     layerId: "your-layer-id",
   });
   ```

**To get features in tile from the interactive map layer:**

The interactive API supports the following tile types:

_web_ - for the Mercator projection (used by OpenStreetMaps, Google Maps, Bing Maps and others) format: level_x_y;
for example: 10_100_100 means level 10. x-coordinate 100, y-coordinate 100.

> _tms_ - for the Tile Map Service developed by the Open Source Geospatial Foundation format: level_x_y; for example: 10_100_100 means level 10 x-coordinate 100, \* y-coordinate 100.
> 
> _quadkey_ - for quadtree keys used by Bing Maps (formerly Virtual Earth) and others format: for example 0123031233 for level 10.
> 
> _here_ - for the HERE tiling schema.
> 
> 1. Create the `RequestBuilder` object.
> 
>    For instructions, see [Create RequestBuilder](#create-requestbuilder).
> 
> 2. Call the `getFeaturesByTile` function with the IDs of the features that you request and layer ID.
> 
>    ```typescript
>    const result = await InteractiveApi.getFeaturesByTile(requestBuilder, {
>      tileType: "your tile type",
>      layerId: "your-layer-id",
>      tileId: "your-tileId",
>      params: "your-params-string",
>    });
>    ```
> 
> **To spatial search for features:**
> 
> 1. Create the `RequestBuilder` object.
> 
>    For instructions, see [Create RequestBuilder](#create-requestbuilder).
> 
> 2. You can use functions `getFeaturesBySpatial` and `getFeaturesBySpatialPost` for a spatial search. For more information, see the [Spatial Search for Features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-spatial.html).
> 
> **To searching for features:**
> 
> 1. Create the `RequestBuilder` object.
> 
>    For instructions, see [Create RequestBuilder](#create-requestbuilder).
> 
> 2. You can use function `searchFeatures` for a searching for features. For more information, see the [Searching for features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-search.html).
> 
> **Iterating features:**
> 
> 1. Create the `RequestBuilder` object.
> 
>    For instructions, see [Create RequestBuilder](#create-requestbuilder).
> 
> 2. You can use function `iterateFeatures` for a searching for features. For more information, see the [Iterating features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-iterate.html).
> 
> **To get statistics:**
> 
> The statistics endpoint provides the number of features in a layer, the size, and a list of properties on the layer.
> 
> 1. Create the `RequestBuilder` object.
> 
>    For instructions, see [Create RequestBuilder](#create-requestbuilder).
> 
> 2. Call the `getStatistics` function with the ID of the layer.
> 
>    ```typescript
>    const result = await InteractiveApi.getStatistics(requestBuilder, {
>      layerId: "your-layer-id",
>    });
>    ```

Reviewer: Let's make it shorter.
```suggestion
## Get data from an interactive map layer

Each [interactive map layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers) has a different set of features that you can request and use to work with map data.

You can request one or several features of the interactive map layer using their IDs, tiles that contain them, a bounding box, spatial search, search, iteration, and statistics.

> #### Note
> When you request features from the interactive map layer, you get `FeatureCollection` even if the request returns only one feature or there are no features found.

The interactive API supports the following tile types:

- `web` – for the Mercator projection (used by OpenStreetMap, Google Maps, and Bing Maps). Format: `level_x_y`. For example, `10_100_100` means level 10, x-coordinate 100, and y-coordinate 100.

- `tms` – for the Tile Map Service specification developed by the Open Source Geospatial Foundation. Format: `level_x_y`. For example, `10_100_100` means level 10, x-coordinate 100, y-coordinate 100.

- `quadkey` – for quadtree keys used by Bing Maps (formerly Virtual Earth). For example, 0123031233 is a quadkey for the level 10 tile.

- `here` – for the HERE tiling schema.

**To get features from the interactive layer:**

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

2. Depending on what you want to use to get features, do one of the following:

    - To get one feature using its ID, call the `getFeature` function with the request builder, feature ID, and layer ID.

      ```typescript
      const result = await InteractiveApi.getFeature(requestBuilder, {
        id: "your-feature-id",
        layerId: "your-layer-id",
      });
      ```

    - To get more than one feature using their IDs, call the `getFeatures` function with the request builder, feature IDs, and layer ID.

      ```typescript
      const result = await InteractiveApi.getFeatures(requestBuilder, {
        id: "your-feature1-id,your-feature2-id",
        layerId: "your-layer-id",
      });
      ```

    - To get features using a bounding box, call the `getFeaturesByBBox` function with the request builder, feature IDs, and layer ID.

      ```typescript
      const result = await InteractiveApi.getFeaturesByBBox(requestBuilder, {
          bbox: "13.082,52.416,13.628,52.626", // Bounding box of Berlin
          layerId: "your-layer-id",
      });
      ```

    - To get features using a tile, call the `getFeaturesByTile` function with the request builder, feature IDs, and layer ID.

      ```typescript
      const result = await InteractiveApi.getFeaturesByTile(requestBuilder, {
        tileType: "your tile type",
        layerId: "your-layer-id",
        tileId: "your-tileId",
        params: "your-params-string",
      });
      ```

    - To get features using the spatial search, call the `getFeaturesBySpatial` or `getFeaturesBySpatialPost` functions.

      With the spatial search, you can find all features around a given position or in a given region. For more information, see [Spatial search for features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-spatial.html) in the Data API Guide.

    - To get features using the search, call the `searchFeatures` function.

      For more information, see [Searching for features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-search.html) in the Data API Guide.

    - To get features using iteration, call the `iterateFeatures` function.

      You get an ordered set of features, and none of them is returned twice. For more information, see [Iterating features](https://developer.here.com/documentation/data-api/data_dev_guide/rest/getting-interactive-iterate.html) in the Data API Guide.

    - To get features using statistics, call the `getStatistics` function with the ID of the layer.

      ```typescript
      const result = await InteractiveApi.getStatistics(requestBuilder, {
        layerId: "your-layer-id",
      });
      ```
```

> **Update Data in an Interactive Map Layer:**
> 
> 1. Create the `RequestBuilder` object.
> 
>    For instructions, see [Create RequestBuilder](#create-requestbuilder).
> 
> 2. You can update the features in an interactive map layer by using functions `patchFeature`, `postFeatures`, `putFeature`, `putFeatures`. For more information, see the [Update Data in an Interactive Map Layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/updating-data-interactive.html) and [Publish to an Interactive Map Layer
>    ](https://developer.here.com/documentation/data-api/data_dev_guide/rest/publishing-data-interactive.html).

Reviewer: ```suggestion
## Update data in and publish it to an interactive map layer

You can update existing features in an interactive map layer or publish new ones.

**To update and publish features:**

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

2. Call the `patchFeature`, `postFeatures`, `putFeature`, or `putFeatures`  functions.
    For more information, see [Update data in an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/updating-data-interactive.html) and [Publish to an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/publishing-data-interactive.html) in the Data API Guide.
```

> **Delete Data from an Interactive Map Layer:**
> 
> 1. Create the `RequestBuilder` object.
> 
>    For instructions, see [Create RequestBuilder](#create-requestbuilder).
> 
> 2. You can update the features in an interactive map layer by using functions `deleteFeature`, `deleteFeatures`. For more information, see the [Delete Data from an Interactive Map Layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/deleting-data-interactive.html).
> 
> ---
> 
> For more information, see the [Data User Guide](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html) and [Interactive API](https://developer.here.com/documentation/data-api/api-reference-interactive.html).

Reviewer: ```suggestion
## Delete data from an interactive map layer

You can delete features from an interactive map layer when you no longer need them.

1. Create the `RequestBuilder` object.

   For instructions, see [Create RequestBuilder](#create-requestbuilder).

2. Call the `deleteFeature` or `deleteFeatures` functions.
    For more information, see [Delete data from an interactive map layer](https://developer.here.com/documentation/data-api/data_dev_guide/rest/deleting-data-interactive.html) in the Data API Guide.
```
